File: backend/application/src/main/java/ua/com/application/models/NBAPlayer.java
package ua.com.application.models;

import lombok.*;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

@Data
@AllArgsConstructor
@NoArgsConstructor
@ToString
@EqualsAndHashCode
@Document(collection = "players")
public class NBAPlayer {
    @Id
    private String id;
    private String name;
    private Team team;
    private double pointsPerGame;
    private int gamesPlayed;
    private double overallScoringPercentage;
    private double threePointPercentage;
    private double rebounds;
    private double steels;

}

File: frontend/application-frontend/src/app/models/nbaplayer.ts
import {Team} from "./team";

export interface NBAPlayer {

  // constructor(name: string, team: Team, pointsPerGame: number, gamesPlayed: number, overallScoringPercentage: number, threePointPercentage: number, rebounds: number, steels: number) {
  //   this.name = name;
  //   this.team = team;
  //   this.pointsPerGame = pointsPerGame;
  //   this.gamesPlayed = gamesPlayed;
  //   this.overallScoringPercentage = overallScoringPercentage;
  //   this.threePointPercentage = threePointPercentage;
  //   this.rebounds = rebounds;
  //   this.steels = steels;
  // }

  id?: string;
  name: string;
  team: Team;
  pointsPerGame: number;
  gamesPlayed: number;
  overallScoringPercentage: number;
  threePointPercentage: number;
  rebounds: number;
  steels: number;
}
